compute/client_library/ingredients/disks/attach_disk.py

```diff
@@ -0,0 +1,57 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# This is an ingredient file. It is not meant to be run directly. Check the samples/snippets
+# folder for complete code samples that are ready to be used.
+# Disabling flake8 for the ingredients file, as it would fail F821 - undefined name check.
+# flake8: noqa
+
+from typing import NoReturn
+
+from google.cloud import compute_v1
+
+
+# <INGREDIENT attach_disk>
+def attach_disk(project_id: str, zone: str, instance_name: str, disk_link: str, mode: str = "READ_ONLY") -> NoReturn:
+    """
+    Attaches a disk resource to a specified compute instance. The disk might be zonal or regional.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone:name of the zone in which the instance you want to use resides.
+        instance_name: name of the compute instance you want to attach a disk to.
+        disk_link: full link to the disk you want to attach. This can be either regional or zonal disk.
+            Expected formats:
+                * https://www.googleapis.com/compute/v1/projects/[project]/zones/[zone]/disks/[disk_name]
+                * /projects/[project]/zones/[zone]/disks/[disk_name]
+                * /projects/[project]/regions/[region]/disks/[disk_name]
+        mode: Specifies in what mode the disk will be attached to the instance. Available options are `READ_ONLY`
+            and `READ_WRITE`. Disk in `READ_ONLY` mode can be attached to multiple instances at once.
+    """
```

Comment:
WDYT about adding links to IAM permissions required to run this sample? https://cloud.google.com/compute/docs/disks/regional-persistent-disk#expandable-1 in this case.

Reply:
Sure, good idea.